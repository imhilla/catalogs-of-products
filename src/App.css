.pokeContainer {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: auto;
}

.container-div {
  padding: 0.5em;
}

.img-container {
  background-color: gray;
  padding: 0.3em;
}

.container-div img {
  width: 100%;
  height: auto;
}

.item-name {
  font-size: 18px;
}