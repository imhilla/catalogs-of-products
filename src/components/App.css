.LowerNav {
  display: flex;
  justify-content: center;
  padding: 1em 0;
}

.lii {
  font-weight: bold;
}
