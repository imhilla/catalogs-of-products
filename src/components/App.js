import React from 'react';
// import PropTypes from 'prop-types';

import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link
} from "react-router-dom";
import Products from './Products';
import PokemonView from './PokemonView'
import '../App.css';
import Pokemon from './pokemon';

function App() {

  return (
    <Router>
      <div>
        <nav>
          <ul>
            <li>
              <Link to="/">Products</Link>
            </li>
          </ul>
        </nav>
        <Switch>
          <Route path="/" exact component={Products} />
          <Route path="/pokemon/:pokemon_id" component={PokemonView} />
        </Switch>
      </div>
    </Router>
  );
}

export default App;
