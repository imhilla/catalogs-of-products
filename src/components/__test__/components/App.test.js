import React from 'react';
import { shallow, mount } from 'enzyme';
import thunk from 'redux-thunk';
import { MemoryRouter } from 'react-router';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
// import Products from 'src/containers/Products';
import App from '../../App';
// import 'src/setupTests';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});

beforeEach(() => {
  store.clearActions();
});

describe('<App />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<App />);
      const component = wrapper.dive();
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});

// describe('test of paths', () => {
//   test('valid path should redirect to products', () => {
//     const wrapper = mount(
//       <MemoryRouter initialEntries={['/']}>
//         <Provider store={store}>
//           <Products />
//         </Provider>

//       </MemoryRouter>,
//     );
//     expect(wrapper.find(Products).toHaveLength(1);
//   });

//   test('valid path should redirect to mission id,', () => {
//     const wrapper = mount(
//       <MemoryRouter initialEntries={['/mission/1']}>
//         <Provider store={store}>
//           <Mission match={{ params: { ID: 1 } }} />
//         </Provider>
//       </MemoryRouter>,
//     );

//     expect(wrapper.find(Mission)).toHaveLength(1);
//   });
// });
