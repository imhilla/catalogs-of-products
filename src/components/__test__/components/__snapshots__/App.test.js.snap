// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render() renders the component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div>
    <nav>
      <div
        className="LowerNav"
      >
        <Link
          className="lii"
          to="/"
        >
          HOME
        </Link>
        <h2>
          SERVICES
        </h2>
        <h2>
          SUPPORT
        </h2>
        <h2>
          CONTACT
        </h2>
        <h2>
          CART
        </h2>
      </div>
      <ul>
        <li />
      </ul>
    </nav>
    <Switch>
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/pokemon/:pokemon_id"
      />
    </Switch>
  </div>
</Router>
`;
